src/runtime/pkg/containerd-shim-v2/create.go

```diff
@@ -103,6 +103,20 @@ func create(ctx context.Context, s *service, r *taskAPI.CreateTaskRequest) (*con
 		s.ctx = newCtx
 		defer span.End()
 
+		// Sandbox sizing information *may* be provided in two scenarios:
+		//   1. The upper layer runtime (ie, containerd or crio) provide sandbox sizing information as an annotation
+		//	in the 'sandbox container's' spec. This would typically be a scenario where as part of a create sandbox
+		//	request the upper layer runtime receives this information as part of a pod, and makes it available to us
+		//	for sizing purposes.
+		//   2. If this is not a sandbox infrastructure container, but instead a standalone single container (analogous to "docker run..."),
+		//	then the container spec itself will contain appropriate sizing information for the entire sandbox (since it is
+		//	a single container.
+		if containerType == vc.PodSandbox {
+			s.config.SandboxCPUs, s.config.SandboxMemMB = oci.CalculateSandboxSizing(ociSpec)
```

Comment:
What if there is no upper layer annotation for sandbox sizing, and `static_sandbox_resource_mgmt` is enabled?

Reply:
For SW/HW architectures w/ out hotplug, this'll behave the same. The VM will be sized based on default CPU/memory from the toml, and won't be adjusted after boot.